buildspec/release.yaml

```
@@ -1,18 +1,18 @@
 
 version: 0.2
 phases:
-  pre_build:
-    commands:
-      - echo Logging in to Amazon ECR...
-      - $(aws ecr get-login --no-include-email --region $AWS_DEFAULT_REGION)
   build:
     commands:
+      - echo Logging in to Amazon ECR...
+      - aws ecr get-login-password --region ${AWS_DEFAULT_REGION} | docker login --username AWS --password-stdin ${REPOSITORY_URI}
       - echo Building the Docker image...
       - PATCH_TAG=${TAG_NAME#"v"} # major.minor.patch
       - MINOR_TAG=${PATCH_TAG%.*} # major.minor
       - MAJOR_TAG=${MINOR_TAG%.*} # major
       - docker build -t ${REPOSITORY_URI}:${PATCH_TAG} -t ${REPOSITORY_URI}:${MINOR_TAG} -t ${REPOSITORY_URI}:${MAJOR_TAG} .
-      - docker push ${REPOSITORY_URI}
+      - docker push ${REPOSITORY_URI}:${PATCH_TAG}
+      - docker push ${REPOSITORY_URI}:${MINOR_TAG}
+      - docker push ${REPOSITORY_URI}:${MAJOR_TAG}
```

Now that this is on the new codebuild image with docker 20.10 you can do
```suggestion
      - docker image push --all-tags  ${REPOSITORY_URI}
```